features/yield/FarmingCalculator.tsx

```diff
@@ -158,10 +175,32 @@ const FarmingCalculator = () => {
     };
   };
 
+  // Set roll and LM reward token information
+  useEffect(() => {
+    if (address) {
+      setTokenAddress(address.toLowerCase());
+    }
+    if (Object.keys(WEEKLY_UMA_REWARDS).includes(tokenAddress)) {
+      setRewardToken(WEEKLY_UMA_REWARDS[tokenAddress]);
+    }
+
+    // Check if token should display roll information.
+    if (Object.keys(ROLL_REWARDS_SCHEDULE).includes(tokenAddress)) {
+      setRollFromTokenObj(ROLL_REWARDS_SCHEDULE[tokenAddress]);
+    } else {
```

Comment:
This is the main change here. `rollFromTokenObj` is set to non-null if there is mapping in the `ROLL_REWARDS_SCHEDULE` struct for the current token address (`tokenAddress`)